tests/compose-tests/test-boot-location-modules.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -xeuo pipefail
+
+dn=$(cd $(dirname $0) && pwd)
+. ${dn}/libcomposetest.sh
+
+prepare_compose_test "bootlocation-modules"
+pysetjsonmember "boot_location" '"modules"'
+runcompose
+echo "ok compose"
+
+# Nothing in /boot (but it should exist)
+ostree --repo=${repobuild} ls -R ${treeref} /boot > bootls.txt
+cat >bootls-expected.txt <<EOF
+d00755 0 0      0 /boot
+EOF
+diff -u bootls{-expected,}.txt
+# Verify /usr/lib/ostree-boot
+ostree --repo=${repobuild} ls -R ${treeref} /usr/lib/ostree-boot > bootls.txt
```

Comment:
Minor: do we even need the directory to exist at all? It'd be cleaner to not even have it at all, no?

Reply:
For systems installed by Anaconda it's where the bootloader data lives.

For FCOS...we could actually inject it statically into `/boot` and drop it from the tree.  That'd have to be a new config option...

Reply:
Or really it'd have to be like `compose tree --bootdir=$(pwd)/tmp/boot` and then we change cosa to drop that stuff in the disk image's `/boot` - after we stop using Anaconda.

Reply:
Related https://github.com/ostreedev/ostree/issues/717#issuecomment-384364150
and https://github.com/projectatomic/rpm-ostree/pull/969